newdle/schemas.py

```diff
@@ -26,13 +33,20 @@ def sort_users(self, data, many, **kwargs):
         return data
 
 
-class NewParticipantSchema(mm.Schema):
+class NewUnknownParticipantSchema(mm.Schema):
     name = fields.String(required=True)
-    email = fields.String()
-    auth_uid = fields.String()
 
 
-class ParticipantSchema(NewParticipantSchema):
+class NewKnownParticipantSchema(NewUnknownParticipantSchema):
+    email = fields.String(required=True)
+    auth_uid = fields.String(required=True)
+    signature = fields.String(required=True)
```

Comment:
I think the signature check could be done in a [`validates_schema`](https://marshmallow.readthedocs.io/en/stable/api_reference.html#marshmallow.decorators.validates_schema)-decorated function here

Reply:
Great idea, much cleaner!